xrpl/transaction/amm_clawback.go

```diff
@@ -0,0 +1,129 @@
+package transaction
+
+import (
+	"errors"
+
+	addresscodec "github.com/Peersyst/xrpl-go/address-codec"
+	"github.com/Peersyst/xrpl-go/xrpl/transaction/types"
+)
+
+const (
+	// Claw back the specified amount of Asset, and a corresponding amount of Asset2 based on
+	// the AMM pool's asset proportion; both assets must be issued by the issuer in the Account
+	// field. If this flag isn't enabled, the issuer claws back the specified amount of Asset,
+	// while a corresponding proportion of Asset2 goes back to the Holder.
+	tfClawTwoAssets uint32 = 0x00000001
+)
+
+var (
+	// ErrInvalidHolder is returned when the holder is invalid.
+	ErrInvalidHolder = errors.New("invalid holder")
+	// ErrInvalidAmountIssuer is returned when the amount issuer is invalid.
+	ErrInvalidAmountIssuer = errors.New("invalid amount issuer")
+)
+
+// Claw back tokens from a holder who has deposited your issued tokens into an AMM pool.
+// Clawback is disabled by default. To use clawback, you must send an AccountSet transaction
+// to enable the Allow Trust Line Clawback setting. An issuer with any existing tokens cannot
+// enable clawback. You can only enable Allow Trust Line Clawback if you have a completely empty
+// owner directory, meaning you must do so before you set up any trust lines, offers, escrows,
+// payment channels, checks, or signer lists. After you enable clawback, it cannot reverted:
+// the account permanently gains the ability to claw back issued assets on trust lines.
+// (Added by the AMMClawback amendment)
+//
+// ```json
+//
+//	{
+//	  "TransactionType": "AMMClawback",
+//	  "Account": "rPdYxU9dNkbzC5Y2h4jLbVJ3rMRrk7WVRL",
+//	  "Holder": "rvYAfWj5gh67oV6fW32ZzP3Aw4Eubs59B",
+//	  "Asset": {
+//	      "currency" : "FOO",
+//	      "issuer" : "rPdYxU9dNkbzC5Y2h4jLbVJ3rMRrk7WVRL"
+//	  },
+//	  "Asset2" : {
+//	      "currency" : "BAR",
+//	      "issuer" : "rHtptZx1yHf6Yv43s1RWffM3XnEYv3XhRg"
+//	  },
+//	  "Amount": {
+//	      "currency" : "FOO",
+//	      "issuer" : "rPdYxU9dNkbzC5Y2h4jLbVJ3rMRrk7WVRL",
+//	      "value" : "1000"
+//	  }
+//	}
+//
+// ```
+type AMMClawback struct {
+	BaseTx
+	// The account holding the asset to be clawed back.
+	Holder string
+	// Specifies the asset that the issuer wants to claw back from the AMM pool.
+	// The asset can be XRP, a token, or an MPT (see: Specifying Without Amounts).
+	// The issuer field must match with Account.
+	Asset types.IssuedCurrency
```

Comment:
We have this struct that we already use in `AmmDeposit`:

```
type Asset struct {
	Currency string        `json:"currency"`
	Issuer   types.Address `json:"issuer,omitempty"`
}
```

I believe we should use it again no?

Reply:
No, as you can see https://github.com/XRPLF/xrpl.js/blob/9211ff6e48fd919c2a90716e856c1a5fe305cfba/packages/xrpl/src/models/common/index.ts#L12 the issuer must not be treated as optional